community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/UpdateGraph.scala

```diff
@@ -193,7 +197,7 @@ case class UpdateGraph(mutatingPatterns: Seq[MutatingPattern] = Seq.empty) {
    * and what is deleted here
    */
   def deleteOverlap(qg: QueryGraph): Boolean = {
-    val identifiersToRead = qg.patternNodes ++ qg.patternRelationships.map(_.name)
+    val identifiersToRead = qg.patternNodes ++ qg.patternRelationships.map(_.name) ++ qg.argumentIds
```

Comment:
@henriknyman This is what you suggested when we were looking into the `UNWIND` cases -- perhaps it is the correct thing after all.